lib/util/util.js

```diff
@@ -119,40 +119,40 @@ const Util = {
      *
      * @param {object} properties javascript object in .mcdevrc.json
      * @param {boolean} [silent] set to true for internal use w/o cli output
-     * @returns {boolean|String[]} file structure ok OR list of fields to be fixed
+     * @returns {Promise<boolean|String[]>} file structure ok OR list of fields to be fixed
      */
-    checkProperties: function (properties, silent) {
-        if (!fs.existsSync(Util.configFileName) || !properties) {
+    checkProperties: async function (properties, silent) {
+        if (!(await fs.pathExists(Util.configFileName)) || !properties) {
             Util.logger.error(`\nCould not find ${Util.configFileName} in ${process.cwd()}.`);
             Util.logger.error(`Run 'mcdev init' to initialize your project.\n`);
             return false;
         }
-        if (!fs.existsSync(Util.authFileName) || !properties) {
+        if (!(await fs.pathExists(Util.authFileName)) || !properties) {
             Util.logger.error(`\nCould not find ${Util.authFileName} in ${process.cwd()}.`);
             Util.logger.error(`Run 'mcdev init' to initialize your project.\n`);
             return false;
         }
 
-        // check if user is running older mcdev version than whats saved to the config
-        if (properties.version && semver.gt(properties.version, packageJsonMcdev.version)) {
-            // dont run this for Catalyst to MC DevTools migration
+        // check if user is running older (ignores patches) mcdev version than whats saved to the config
+        if (
+            properties.version &&
+            ![null, 'patch'].includes(semver.diff(properties.version, packageJsonMcdev.version))
```

Comment:
ok this is doing the limitation on patch which is cool. but we also need the gt() together with this (&&) to check if MY APP needs upgrading because the config is newer.

Reply:
resolved in discussion